dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -0,0 +1,756 @@
+    private static Set<String> locatorsFileContent = new LinkedHashSet<String>();
+    private static Set<String> leadsFileContent = new LinkedHashSet<String>();
+    private static Set<String> snappyJobLogFilesForTask = new LinkedHashSet<String>();
+    private static Set<String> snappyJobLogFilesForCloseTask = new LinkedHashSet<String>();
+    private static Set<Integer> pids = new LinkedHashSet<Integer>();
+    private static String locatorsFilePath = null;
+    private static String serversFilePath = null;
+    private static String leadsFilePath = null;
+    private static String userAppJar = null;
+    private static String simulateStreamScriptName = TestConfig.tab().stringAt(SnappyPrms.simulateStreamScriptName, "simulateFileStream");
+    private static String simulateStreamScriptDestinationFolder = null;
+    private static String leadHost = null;
+    public static Long waitTimeBeforeJobStatusInTask = TestConfig.tab().longAt(SnappyPrms.jobExecutionTimeInMillisForTask, 6000);
+    public static Long waitTimeBeforeStreamingJobStatusInTask = TestConfig.tab().longAt(SnappyPrms.streamingJobExecutionTimeInMillisForTask, 6000);
+    public static Long waitTimeBeforeJobStatusInCloseTask = TestConfig.tab().longAt(SnappyPrms.jobExecutionTimeInMillisForCloseTask, 6000);
+
+
+    public static <A, B> Map<A, B> toScalaMap(HashMap<A, B> m) {
+        return JavaConverters.mapAsScalaMapConverter(m).asScala().toMap(Predef.<Tuple2<A, B>>conforms());
+    }
+
+    public static void HydraTask_initializeSnappy() {
+        SparkContext sc = null;
+        snc = SnappyContext.getOrCreate(sc);
+        Log.getLogWriter().info("SnappyContext initialized successfully");
+    }
+
+
+    public static void HydraTask_stopSnappy() {
+        snc.stop();
+        Log.getLogWriter().info("SnappyContext stopped successfully");
+    }
+
+    protected String getDataLocation() {
+        String scriptPath = null;
+        try {
+            hd = getHostDescription();
+            scriptPath = hd.getGemFireHome() + sep + ".." + sep + "snappy"
+                    + sep + "quickstart" + sep + "data";
+        } catch (Exception e) {
+            throw e;
+        }
+        return scriptPath;
+    }
+
+    protected HostDescription getHostDescription() {
+        hd = TestConfig.getInstance()
+                .getClientDescription(RemoteTestModule.getMyClientName())
+                .getVmDescription().getHostDescription();
+        sep = hd.getFileSep();
+        return hd;
+    }
+
+    protected String getScriptLocation(String scriptName) {
+        String scriptPath = null;
+        try {
+            hd = getHostDescription();
+            scriptPath = hd.getGemFireHome() + sep + ".." + sep + "snappy"
+                    + sep + "sbin" + sep + scriptName;
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
```

Comment:
Not Sure what exception you are expecting here but still the only thing that can throw an exception is getHostDescription so please add the required Exception handling inside this method and remove from everywhere else